tap-version-14-specification.md

```diff
@@ -0,0 +1,903 @@
+the Test Point ID 4 is outside the stated Plan range.
+
+```tap
+TAP version 14
+1..3
+ok 2
+ok 4
+ok 1
+```
+
+Test Point IDs _should_ be unique within the TAP Document.  Harnesses _may_
+warn about repeated Test Point IDs or treat them as a test failure, but
+_must not_ treat a Test Point with a re-used ID as a non-TAP line.
+
+#### Description
+
+Any text after the test number but before a `#` is the description of
+the test point.
+
+```tap
+ok 42 - this is the description of the test
+```
+
+Descriptions _should_ be separated from the Test Point Status and Test
+Point ID by the string `" - "`, in order to prevent confusing a numeric
+description with a Test Point ID.  Harnesses _must_ treat Test Points
+identically whether the description starts with `" - "` or not.
+
+For example, these two test points _must_ be treated identically by a
+Harness:
+
+```tap
+ok 1 this is fine
+ok 1 - this is fine
+```
+
+Harnesses _should not_ consider a leading `" - "` to be a part of the
+description reported to a user.
+
+#### Directive
+
+Directives are special notes that follow the first unescaped `#` on the
+Test Point line that is preceded and followed by whitespace.  Only two are
+currently defined: `TODO` and `SKIP`.
+
+Directives are not case sensitive.  That is, Harnesses _must_ treat `#
+SKIP`, `# skip`, and `# SkIp` identically.
+
+Harnesses _may_ support additional platform-specific Directives.  Future
+versions of this specification may codify additional Directives with
+defined semantics.
+
+Unrecognized Directives _must not_ be treated as test failure, or an
+invalid TAP line.  Harnesses _should_ include any unrecognized directives
+in the Test Point description.
+
+Note that escaped `#` characters are not to be treated as delimiters for
+Directives.  See "Escaping" below.
+
+##### Whitespace Around Directive Delimiter
+
+Earlier versions of the TAP specification were not explicit about
+whitespace requirements regarding directives.  The following rules maximize
+compatibility between TAP14 producers and harnesses:
+
+1. Producers _must_ output directives with at least one space character
+   preceding the `#` in a directive, as well as at least one space
+   character between the `#` and the directive name.
+
+2. Harnesses _must not_ treat escaped `#` characters as directive
+   delimiters.
+
+3. Harnesses _may_ accept directive delimiters where the `#` is not
+   preceded by whitespace, but _should_ warn that such output is
+   non-conformant with the TAP14 specification.
+
+If harnesses choose to parse directives without whitespace before and after
+the `#`, then they ought to consider the impact if test descriptions
+contain URLs and/or may be coming from TAP producers that are not diligent
+about escaping `#` characters.  This should be done only to the extent
+necessary for backwards compatibility with existing TAP producers.
+
+For example:
+
+```tap
+TAP version 13
```

Comment:
`version 14`?

Reply:
Ah, yeah, that's confusing.  The only "version 13" should be in the bit where we say that version 13 is also fine.